addons/format_lame.c

```diff
@@ -0,0 +1,559 @@
+/*
+ * Asterisk -- An open source telephony toolkit.
+ *
+ * Copyright (C) 2024, Naveen Albert, sponsored by Marfox Ltd.
+ * Copyright (C) 2024, Sperl Viktor
+ * 
+ * Naveen Albert <asterisk@phreaknet.org>
+ * Sperl Viktor <viktike32@gmail.com>
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+
+/*!
+ * \file
+ * \brief MP3 Format including encoder and decoder
+ * 
+ * \author Naveen Albert <asterisk@phreaknet.org>
+ * \author Sperl Viktor <viktike32@gmail.com>
+ * 
+ * \ingroup formats
+ */
+
+/*** MODULEINFO
+	<depend>lame</depend>
+	<conflict>format_mp3</conflict>
+	<defaultenabled>no</defaultenabled>
+	<support_level>extended</support_level>
+ ***/
+
+#include "asterisk.h"
+
+#include <lame/lame.h>
+
+#include "asterisk/module.h"
+#include "asterisk/mod_format.h"
+#include "asterisk/logger.h"
+#include "asterisk/format_cache.h"
+
+#define SLIN_BUFLEN 320
+#define DECORDER_OUTLEN 32768
+#define BITS 2
+
+struct mp3lame_private {
+	lame_global_flags * lgfp;
+	hip_global_flags * hgfp;
+
+	int sample_rate;
+
+	/* decoder buffer */
+	short int doutput[DECORDER_OUTLEN *  1024];
+	long doffset;
+	int dsamples;
+
+	/* decoder positions */
+	int total_bytes_decoded, total_bytes_compressed;
+
+	/* counters */
+	unsigned int decoder_counter, encoder_counter;
+};
+
+static void mp3lame_error(const char *format, va_list ap)
+{
+	ast_log_ap(AST_LOG_WARNING, format, ap);
```

Comment:
These ast_log_ap calls aren't compiling for me on Ubuntu 20

Reply:
These were inherited from #665. Altough these looked suspicious, but it did compile for me. I've rewrote these, please check now. Possible, the issue was the missing <stdarg.h> for the va_lists, which I also added.